benchmark/kostya/ondemand.h

```diff
@@ -56,10 +56,10 @@ simdjson_really_inline bool OnDemand::Run(const padded_string &json) {
   count = 0;
 
   auto doc = parser.iterate(json);
-  for (ondemand::object coord : doc["coordinates"]) {
-    sum.x += double(coord["x"]);
-    sum.y += double(coord["y"]);
-    sum.z += double(coord["z"]);
+  for (ondemand::object coord : doc.find_field("coordinates")) {
```

Comment:
@jkeiser These are just direct semantic transposition of the previous code, but should we not also include a version with the order-insensitive code... because that's where the performance differential should appear... No? (I think that the numbers would be great to have.)

Reply:
@lemire I agree, but I'd like to make a benchmark that includes *input* with different order as well, to match. Having more benchmarks is going to be an important part of a release.

Reply:
While I *do* want to do that, I went ahead and added the unordered version for comparison purposes. Unordered still seems to be slightly worse than ordered, by roughly 2%, and it is a real difference caused almost entirely by increased instruction count:

| Benchmark | Throughput | Instructions | Branch Misses |
|---|---|---|---|
| LargeRandom<OnDemand> | 634.066 MB/s | 614,119,494 | 972,480 |
| LargeRandom<OnDemandUnordered> | 650.795 MB/s | 636,119,541 | 956,152 |